test/pstl_offload/memory/system_allocations.h

```diff
@@ -0,0 +1,120 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _SYSTEM_ALLOCATIONS_H
+#define _SYSTEM_ALLOCATIONS_H
+
+#define _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL
+#include <new>
+#define _ONEDPL_PSTL_OFFLOAD_TOP_LEVEL
+#include <cstdlib>
+#include <malloc.h>
+
+static void check_true(bool expected, bool condition, const char* file, int line, const char* message) {
+    if (condition != expected) {
+        fprintf(stderr, "error at %s:%d - %s\n", file, line, message);
+        exit(1);
+    }
+}
+
+#ifndef EXPECT_TRUE
+#define EXPECT_TRUE(condition, message) check_true(true, condition, __FILE__, __LINE__, message)
```

Comment:
I guess we can include `support/utils.h`  under the guard and remove these assertions as well

Reply:
Done.